src/amazonica/aws/kinesis.clj

```diff
@@ -105,9 +105,11 @@
 
 (defn marshall
   [deserializer ^Record record]
-  {:sequence-number (.getSequenceNumber record)
-   :partition-key   (.getPartitionKey record)
-   :data            (deserializer (.getData record))})
+  {:approximate-arrival-timestamp (.getApproximateArrivalTimestamp record)
```

Comment:
To be consistent with the rest of the library I think you would need to manually call `amazonica.core/marshall` on the returned value of `.getApproximateArrivalTimestamp`. `java.util.Date`s are converted to Joda `DateTime`s everywhere else.